src/FastConv/FastConv.cpp

```diff
@@ -12,21 +12,190 @@ CFastConv::~CFastConv( void )
 
 Error_t CFastConv::init(float *pfImpulseResponse, int iLengthOfIr, int iBlockLength /*= 8192*/, ConvCompMode_t eCompMode /*= kFreqDomain*/)
 {
+    if (iLengthOfIr < 1) {
+        return Error_t::kFunctionInvalidArgsError;
+    }
 
+    m_ConvType = eCompMode;
+    m_pfImpulseResponse = new float [iLengthOfIr];
+
+    m_pfInputTail = new float[(2 * iLengthOfIr) - 1];
```

Comment:
You may want to define buffers in a separate class for freq vs time domain, as said by Vedant. This way each buffer's size makes more sense